packages/venia-concept/src/components/ProductOptions/swatch.js

```diff
@@ -1,38 +1,67 @@
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import { bool, number, oneOfType, shape, string } from 'prop-types';
 
 import classify from 'src/classify';
+import Icon from 'src/components/Icon';
 import defaultClasses from './swatch.css';
 
+// TODO: replace with actual swatch colors or images from API
+// M2 GraphQL doesn't currently support them
+const cache = new Map();
+const memoize = fn => key =>
+    cache.has(key) ? cache.get(key) : cache.set(key, fn(key)).get(key);
+
+const getRandomColor = () =>
+    Array.from({ length: 3 }, () => Math.floor(Math.random() * 255)).join(',');
+const memoizedGetRandomColor = memoize(getRandomColor);
+
 class Swatch extends Component {
     static propTypes = {
-        classes: PropTypes.shape({
-            root: PropTypes.string
+        classes: shape({
+            root: string
         }),
-        item: PropTypes.shape({
-            id: PropTypes.string,
-            name: PropTypes.string
+        hasFocus: bool,
+        isSelected: bool,
+        item: shape({
+            id: oneOfType([number, string]),
+            label: string
         })
     };
 
+    get icon() {
+        const { isSelected } = this.props;
+
+        return isSelected ? <Icon name="check" /> : null;
+    }
+
     render() {
-        const { classes, item } = this.props;
-        const { id, name } = item;
-        const style = { '--swatch-color': id };
+        const { icon, props } = this;
+        const {
+            classes,
+            hasFocus, // eslint-disable-line
```

Comment:
This is silly. We need to update our linter config so unused destructured props are acceptable when rest props are being declared.

Reply:
I sympathize, because it feels fast and simple to use a language-native way of blacklisting properties when copying an object. But I don't think we should use this idiom. It's very unintuitive to the newbie. Instead, can we pick a subset of props to pass?

Reply:
Sure.

Reply:
To clarify, we obviously won't whitelist which props get passed to the `button` element; that's why we're using rest props in the first place. I'll explicitly exclude `hasFocus`, though.